src/ModelContextProtocol/Configuration/McpServerBuilderExtensions.cs

```diff
@@ -242,6 +243,119 @@ where t.GetCustomAttribute<McpServerPromptTypeAttribute>() is not null
+        this IMcpServerBuilder builder,
+        McpServerResource resource)
+    {
+        builder.Services.Configure<McpServerOptions>(s =>
+        {
+            var capabilities = s.Capabilities ??= new();
+            var resources = capabilities.Resources ??= new();
+            var collection = resources.ResourceCollection ??= [];
+            collection.Add(resource);
+        });
+
+        return builder;
+    }
+
+    private static IMcpServerBuilder AddResources(
+        this IMcpServerBuilder builder,
+        IEnumerable<McpServerResource> resources)
+    {
+        foreach (var resource in resources)
+        {
+            builder = builder.AddResource(resource);
+        }
+        return builder;
+    }
+
+    /// <summary>
+    /// Adds a resource to the server's capabilities.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="resource">The resource to add.</param>
+    /// <returns>The <see cref="IMcpServerBuilder"/> instance.</returns>
+    public static IMcpServerBuilder WithResource(
+        this IMcpServerBuilder builder,
+        McpServerResource resource)
+    {
+        Throw.IfNull(builder);
+        Throw.IfNull(resource);
+
+        return builder.AddResource(resource);
+    }
+
+    /// <summary>
+    /// Adds a resource to the server's capabilities.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="fileInfo">The file info of the resource.</param>
+    /// <returns>The <see cref="IMcpServerBuilder"/> instance.</returns>
+    public static IMcpServerBuilder WithResource(
+        this IMcpServerBuilder builder,
+        IFileInfo fileInfo)
+    {
+        Throw.IfNull(builder);
+        Throw.IfNull(fileInfo);
+        
+        var resource = fileInfo.ToResource();
+        return builder.AddResource(resource);
+    }
+
+    /// <summary>
+    /// Adds a collection of resources to the server's capabilities.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="resources">The collection of the resources.</param>
+    /// <returns>The <see cref="IMcpServerBuilder"/> instance.</returns>
+    public static IMcpServerBuilder WithResources(
+        this IMcpServerBuilder builder,
+        params IEnumerable<McpServerResource> resources)
+    {
+        Throw.IfNull(builder);
+        Throw.IfNull(resources);
+
+        return builder.AddResources(resources);
+    }
+
+    /// <summary>
+    /// Adds a collection of resources to the server's capabilities.
+    /// </summary>
+    /// <param name="builder">The builder instance.</param>
+    /// <param name="fileInfos">The collection of the file info of the resources.</param>
+    /// <returns>The <see cref="IMcpServerBuilder"/> instance.</returns>
+    public static IMcpServerBuilder WithResources(
+        this IMcpServerBuilder builder,
+        params IEnumerable<IFileInfo> fileInfos)
```

Comment:
I don't think we should have such IFileInfo overloads, both for the mime type issue, but also because it's not clear exactly when or how we'd be reading the data from the file. Would every request require re-reading the data? If not, what is the caching policy? Would we be required to subscribe to file change notifications in order to raise resource changed notifications?  _Maybe_ at some point we'd want to sign up for all that, but I'd prefer if we do, for it to be via creating McpServerResource instances, with Create methods that are configurable with various details.